netstate/auth/process_api_key.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+package main
+
+import  (
+	"fmt"
+	"os"
+	"flag"
+)
+ 
+type httpHeaders struct {
+	XApiKey string
+	Connection string 
+	CacheControl string
+	UpgradeInsecureRequests string 
+	Accept string
+	AcceptLanguage string
+}
+
+func main() {
+	key := flag.String("key", "", "this is your API KEY")
+	flag.Parse()
+	
+	httpRequestHeaders := createHeaderStruct()
+	if len(*key) == 0 {
+		_setEnv()
+		validateApiKeyWithEnv(httpRequestHeaders)
+	} else {
+		validateApiKeyWithFlags(httpRequestHeaders, key)
+	}
+}
+
+func createHeaderStruct() *httpHeaders {
+	// mock HTTP request headers
+	// client can set headers x-api-key with 
+	//req.Header.Add("x-api-key", "apikey")
+	requestHeader := httpHeaders {
+		XApiKey: "12345",
+		Connection: "keep-alive",
+		CacheControl: "max-age=0",
+		UpgradeInsecureRequests: "1",
+		Accept: "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8" ,
+		AcceptLanguage: "en-US,en;q=0.9",
+	}
+	return &requestHeader
+}
+
+func _setEnv() {
+	os.Setenv("APIKEY","1245")
+}
+
+func validateApiKeyWithEnv(headers* httpHeaders)(bool) {
```

Comment:
Go is super weird! since the first letter of this function name is lower case, it's considered private, and can't be used outside of this package. You need to capitalize the first letter to make sure other libraries can use it

Reply:
fixed! thanks